### PR TITLE
ENH: default EpochsArray events param

### PR DESCRIPTION
**Pull request by kingjr**
closes https://github.com/mne-tools/mne-python/issues/3189

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Don't forget test

File: doc/whats_new.rst [hunk 1]
```diff
@@ -119,6 +119,8 @@ BUG
 
     - Fix bug in source normal adjustment that occurred when 1) patch information is available (e.g., when distances have been calculated) and 2) points are excluded from the source space (by inner skull distance) by `Eric Larson`_
 
+    - Change default scoring method of :func:`mne.decoding.GeneralizationAcrossTime` and :func:`mne.decoding.TimeDecoding` to estimate the scores within the cross-validation as in scikit-learn. The method can be changed with the ``score_mode`` parameter by `Jean-Remi King`_
+
```
Comment (kingjr) [inline doc/whats_new.rst]: whoops, this is supposed to go in https://github.com/mne-tools/mne-python/pull/3143

File: mne/epochs.py [hunk 2]
```diff
@@ -2164,6 +2166,9 @@ def __init__(self, data, info, events, tmin=0, event_id=None,
         if len(info['ch_names']) != data.shape[1]:
             raise ValueError('Info and data must have same number of '
                              'channels.')
+        if events is None:
+            n_trial = len(data)
+            events = np.c_[range(n_trial), [0] * n_trial, [1] * n_trial]
```
Comment (dengemann) [inline mne/epochs.py]: not sure if this goes well with pytho 3.
why not standard numpy:

```
np.c_[np.arange(n_events), np.zeros(n_events), np.ones(n_events)]
```
Also we usually don't call it trials, maybe events or epochs?

**Comment by dengemann** [comment]
Other than my comment looks good!

File: doc/whats_new.rst [hunk 3]
```diff
@@ -156,6 +156,8 @@ API
 
     - The C wrapper :func:`mne.do_forward_solution` has been deprecated in favor of the native Python version :func:`mne.make_forward_solution` by `Eric Larson`_
 
+    - The ``event`` parameter of :func:`mne.epochs.EpochsArray` is set by defaults to chronological time-samples and event values of 1, by `Jean-Remi King`_
```
Comment (Eric89GXL) [inline doc/whats_new.rst]: by default

File: mne/tests/test_epochs.py [hunk 4]
```diff
@@ -1967,6 +1967,13 @@ def test_array_epochs():
     epochs = EpochsArray(data_1, info, events=events_1, event_id=1,
                          tmin=-0.2, baseline=(None, 0))
 
+    # default events
+    epochs = EpochsArray(data_1, info)
+    assert_array_equal(epochs.events[:, 0], range(len(data_1)))
```
Comment (Eric89GXL) [inline mne/tests/test_epochs.py]: probably also needs to be `np.arange`

File: doc/whats_new.rst [hunk 5]
```diff
@@ -156,6 +156,8 @@ API
 
     - The C wrapper :func:`mne.do_forward_solution` has been deprecated in favor of the native Python version :func:`mne.make_forward_solution` by `Eric Larson`_
 
+    - The ``event`` parameter of :func:`mne.epochs.EpochsArray` is set by default to chronological time-samples and event values of 1, by `Jean-Remi King`_
```
Comment (agramfort) [inline doc/whats_new.rst]: event -> events

Comment (agramfort) [inline doc/whats_new.rst]: of 1 -> to 1

File: mne/epochs.py [hunk 6]
```diff
@@ -2164,6 +2166,10 @@ def __init__(self, data, info, events, tmin=0, event_id=None,
         if len(info['ch_names']) != data.shape[1]:
             raise ValueError('Info and data must have same number of '
                              'channels.')
+        if events is None:
+            n_epochs = len(data)
+            events = np.c_[np.arange(n_epochs), np.zeros(n_epochs),
+                           np.zeros(n_epochs)]
```
Comment (agramfort) [inline mne/epochs.py]: np.ones(n_epochs)

File: doc/whats_new.rst (hunk 5 above)
Comment (Eric89GXL) [inline doc/whats_new.rst]: I think "of 1" actually works slightly better here -- removing the intermediate bits it's either the current "event parameter ... is set by default to ... event values of 1" or "event parameter .. is set by default to ... event values to 1". The former reads better to me.

File: mne/epochs.py (hunk 6 above)
Comment (Eric89GXL) [inline mne/epochs.py]: And also `np.zeros(n_epochs, int)`, `np.ones(n_epochs, int)` otherwise this result will be a float array

**Comment by agramfort** [comment]
merged by rebase.

thx @kingjr

**Pull request closed without merging**